### PR TITLE
Removed Dates and Update Status for Continuous Journal

### PR DESCRIPTION
**Pull request by shreyalpandit**
Addresses Issue #1090 so that the continuous journal does not have a value for all dates except start date.

### CODE REVIEW

**Comment by avishkar58** [comment]
Hey @amritpurshotam would you mind reviewing this PR please?

**Review by amritpurshotam — CHANGES_REQUESTED** [review]
Nice work @shreyalpandit !. Just some small changes to clean up the Event construction and some optional changes for each of the `@property` fields. Otherwise looks good :)

File: api/app/events/models.py [hunk 1]
```diff
@@ -70,21 +70,21 @@ def __init__(
...
+        self.application_open = None if event_type == EventType.CONTINUOUS_JOURNAL else application_open
+        self.application_close = None if event_type == EventType.CONTINUOUS_JOURNAL else application_close
+        self.review_open = None if event_type == EventType.CONTINUOUS_JOURNAL else review_open
+        self.review_close = None if event_type == EventType.CONTINUOUS_JOURNAL else review_close
+        self.selection_open = None if event_type == EventType.CONTINUOUS_JOURNAL else selection_open
+        self.selection_close = None if event_type == EventType.CONTINUOUS_JOURNAL else selection_close
+        self.offer_open = None if event_type == EventType.CONTINUOUS_JOURNAL else offer_open
+        self.offer_close = None if event_type == EventType.CONTINUOUS_JOURNAL else offer_close
+        self.registration_open = None if event_type == EventType.CONTINUOUS_JOURNAL else registration_open
+        self.registration_close = None if event_type == EventType.CONTINUOUS_JOURNAL else registration_close
```
[the remark this answers is not in the document]
Comment (shreyalpandit) [reply api/app/events/models.py:87]: That makes sense! I was also thinking of tracking continuous events after defining the event type enum:

```
continuous_event =
{
    EventType.EVENT: False,
    EventType.AWARD: False,
    EventType.CALL: False,
    EventType.PROGRAMME: False,
    EventType.JOURNAL: False,
    EventType.CONTINUOUS_JOURNAL: True
}
```

and then making a subclass of `Event` called `ContinuousEvent` for now that sets the dates to None. 
```
class ContinuousEvent(Event):
    def __init__(self,
                names,
                descriptions,
                start_date,
                key,
                organisation_id,
                email_from,
                url,
                event_type,
                travel_grant,
                miniconf_url=None):
        super().__init__(self,
                        names=names,
                        descriptions=descriptions,
                        start_date=start_date,
                        end_date=None,
                        key=key,
                        organisation_id=organisation_id,
                        email_from=email_from,
                        url=url,
                        application_open=None, ...)
        
    def update(...):
       ...
```

When future event types are added, they can be marked as continuous events or not, and the rest can stay the same?

File: api/app/events/models.py [hunk 2]
```diff
@@ -12,28 +12,42 @@ class EventType(Enum):
...
-
+def check_open(open, close):
+    now = datetime.now()
+    if open and close:
+        return now >= open and now < close
+    elif open:
+        return now >= open
+    elif close:
+        return now < close
+    return True
```
Comment (amritpurshotam) [inline api/app/events/models.py:23]: I like this, much much cleaner than what I suggested!

**Pull request merged**